daml-assistant/daml-helper/src/DA/Daml/Helper/Ledger.hs

```diff
@@ -345,6 +350,80 @@ allocateParty args name = do
         setRequestBodyJSON $ AllocatePartyRequest {identifierHint = text, displayName = text}
   return party
 
+tokenFor :: [L.Party] -> L.LedgerId -> L.Token
+tokenFor parties ledgerId =
+  L.Token $
+  T.unpack $
+  JWT.encodeSigned
+    (JWT.HMACSecret "secret")
+    mempty
+    mempty
+      { JWT.unregisteredClaims =
+          JWT.ClaimsMap $
+          Map.fromList
+            [ ( "https://daml.com/ledger-api"
+              , A.Object $
+                HashMap.fromList
+                  [ ("actAs", A.toJSON $ map (TL.toStrict . L.unParty) parties)
+                  , ("readAs", A.toJSON $ map (TL.toStrict . L.unParty) parties)
+                  , ("ledgerId", A.String $ TL.toStrict $ L.unLedgerId ledgerId)
+                  ])
+            ]
+      }
+
+runLedgerReset :: LedgerFlags -> IO ()
+runLedgerReset flags = do
+  putStrLn $ "Resetting ledger."
+  args <- getDefaultArgs flags
+  reset args
+
+reset :: LedgerArgs -> IO ()
+reset args =
+  case api args of
+    Grpc ->
+      runWithLedgerArgs args $ do
+        parties0 <- L.listKnownParties
+        let parties = [party | L.PartyDetails {party} <- parties0]
+        let parties' = nubSort $ "Alice" : [L.unParty p | p <- parties]
```

Comment:
"Alice" might not be a valid party identifier (not in the sense that "Alice" has not been allocated but that the format of the identifier is wrong). Wouldn't it make sense to just stop execution if there are no parties since no parties can't create any contracts anyway.

Reply:
That's a better idea. I'll add that.